ipv-core/src/main/java/com/ibm/whc/deid/providers/masking/fhir/FHIRMaskingProvider.java

```diff
@@ -1,70 +1,90 @@
 import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import com.ibm.whc.deid.ObjectMapperFactory;
-import com.ibm.whc.deid.providers.masking.AbstractComplexMaskingProvider;
 import com.ibm.whc.deid.providers.masking.ComplexMaskingProvider;
 import com.ibm.whc.deid.providers.masking.MaskingProviderFactory;
+import com.ibm.whc.deid.pruners.GlobalProcessorFactory;
 import com.ibm.whc.deid.shared.exception.KeyedIllegalArgumentException;
 import com.ibm.whc.deid.shared.exception.KeyedRuntimeException;
 import com.ibm.whc.deid.shared.pojo.config.DeidMaskingConfig;
+import com.ibm.whc.deid.shared.pojo.config.GlobalProcessorConfig;
 import com.ibm.whc.deid.shared.pojo.config.json.JsonConfig;
 import com.ibm.whc.deid.shared.pojo.config.json.JsonMaskingRule;
 import com.ibm.whc.deid.shared.pojo.masking.ReferableData;
+import com.ibm.whc.deid.shared.pojo.masking.ReferableNode;
 import com.ibm.whc.deid.utils.log.LogCodes;
 import com.ibm.whc.deid.utils.log.Messages;
-import scala.Tuple2;
 
-public class FHIRMaskingProvider extends AbstractComplexMaskingProvider
-    implements ComplexMaskingProvider {
+public class FHIRMaskingProvider implements ComplexMaskingProvider, Serializable {
 
-  private static final long serialVersionUID = 5945527984023679481L;
+  private static final long serialVersionUID = 1L;
+
+  public static final String DISABLE_TYPES_VALUE = "default";
+
+  private final GlobalProcessorConfig gpConfig;
+  private final Map<String, MaskingProviderBuilder> maskingProviderMap = new HashMap<>();
+  private String keyForType;
 
   public FHIRMaskingProvider(DeidMaskingConfig maskingConfiguration,
       MaskingProviderFactory maskingProviderFactory, String tenantId) {
-    this(maskingConfiguration, maskingConfiguration.getCertificateId(),
-        maskingConfiguration.isDefaultNoRuleResolution(), maskingProviderFactory, "/fhir/",
-        tenantId);
+    this(maskingConfiguration, maskingConfiguration.isDefaultNoRuleResolution(),
+        maskingProviderFactory, "/fhir/", null, tenantId);
   }
 
-  protected FHIRMaskingProvider(DeidMaskingConfig maskingConfiguration, String certificateId,
-      boolean defNoRuleRes, MaskingProviderFactory maskingProviderFactory, String basePathPrefix,
+  public FHIRMaskingProvider(DeidMaskingConfig maskingConfiguration,
+      MaskingProviderFactory maskingProviderFactory, GlobalProcessorConfig gpConfig,
       String tenantId) {
-    super(maskingConfiguration);
+    this(maskingConfiguration, maskingConfiguration.isDefaultNoRuleResolution(),
+        maskingProviderFactory, "/fhir/", gpConfig, tenantId);
+  }
+
+  protected FHIRMaskingProvider(DeidMaskingConfig maskingConfiguration, boolean defNoRuleRes,
+      MaskingProviderFactory maskingProviderFactory, String basePathPrefix,
+      GlobalProcessorConfig gpConfig, String tenantId) {
 
-    this.maskingProviderFactory = maskingProviderFactory;
-    this.certificateId = certificateId;

```

Comment:
Is the CertificateId removed or kept around somewhere else ? It's something that is supposed to tie the masking configuration with the data and audit trail so that we know what "certificate" the config that was used to mask the data. Hopefully it wasn't broken

Reply:
It was never used in this class.  It is used in the REST API entry methods as that is where the audit records, if requested, are generated.  For spark it is used differently - in AEProcCADFEventGenerator I believe.  We don't generate audit records in the Java jar programmatic path, but I think that tha is OK as the enclosing system would likely need to control any audit records.  It was just an unsed variable in this class.